libdevice/device.h

```diff
@@ -0,0 +1,36 @@
+//==--- device.h - device definitions ------------------------*- C++ -*-----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __LIBDEVICE_DEVICE_H__
+#define __LIBDEVICE_DEVICE_H__
+
+#ifdef __cplusplus
+#define EXTERN_C extern "C"
+#else // __cplusplus
+#define EXTERN_C
+#endif // __cplusplus
+
+#ifdef CL_SYCL_LANGUAGE_VERSION
+#ifndef SYCL_EXTERNAL
+#define SYCL_EXTERNAL
```

Comment:
Where SYCL_EXTERNAL is undef'ed?

Reply:
Sorry, I do not understand the question.  Why does it have to be undef'ed?

Reply:
SYCL complaint compiler must define this macro.
Vlad is asking why do we check if it's not defined.

Reply:
Thank you, Alexey.  SYCL_EXTERNAL is undefined for host compilation, i.e. clang *does not* define SYCL_EXTERNAL for host compilation.  For host compilations it is done by defining SYCL_EXTERNAL to <blank> in CL/sycl/detail/defines.hpp.  Since we decided to get rid of the dependencies on SYCL header files, we have to do it explicitly here.

Reply:
My question was really where this macro is undefined to host compilations?

Reply:
It's not defined by default. It's defined by device compiler only.